code/messaging/src/phone/java/com/adobe/marketing/mobile/MessagingInternal.java

```diff
@@ -79,6 +92,9 @@ protected MessagingInternal(final ExtensionApi extensionApi) {
 
         // Init the messaging state
         messagingState = new MessagingState();
+
+        // create a module to get access to the rules engine for adding ODE rules
```

Comment:
do we not have access via `Extension` (that is, through `self` in this class)?

maybe we should discuss with core team if that's the case - seems like we shouldn't need to make a separate `Module` for this

Reply:
i see, yeah we do not have access to the rules engine unless we provide a module or are calling replaceRules from a module object. we will need to talk to core to expose the functionality.